pkg/cosign/remote/remote.go

```diff
@@ -97,3 +108,55 @@ LayerLoop:
 	}
 	return nil, nil
 }
+
+func (r *ro) Replace(signatures oci.Signatures, o oci.Signature) (oci.Signatures, error) {
+	sigs, err := signatures.Get()
+	if err != nil {
+		return nil, err
+	}
+
+	ros := &replaceOCISignatures{Signatures: signatures}
+
+	sigsCopy := make([]oci.Signature, 0, len(sigs))
+	sigsCopy = append(sigsCopy, o)
+
+	if len(sigs) == 0 {
+		ros.attestations = append(ros.attestations, sigsCopy...)
+		return ros, nil
+	}
+
+	for _, s := range sigs {
+		var payloadData map[string]interface{}
+
+		p, err := s.Payload()
+		if err != nil {
+			return nil, fmt.Errorf("could not get payload: %w", err)
+		}
+
+		err = json.Unmarshal(p, &payloadData)
+		if err != nil {
+			return nil, fmt.Errorf("unmarshal payload data: %w", err)
+		}
+
+		if r.predicateURI == payloadData["payloadType"] {
+			fmt.Println("same found")
```

Comment:
We should make sure logging goes to stderr with fmt.FPrintln, and make it a bit more structued/explanatory here. Maybe:

fmt.Fprintln(os.Stderr, "Attestation already present, not adding new one")
fmt.Fprintln(os.Stderr, "Adding new attestation")

Reply:
thank you, I forgot to fix it, it should be okay right now.